Automattic/wp-calypso
# Modularise sites state

This PR is one of many working on modularising state in Calypso. This one handles sites.

For mode details on state modularization, see #39261 and p4TIVU-9lM-p2.

Unfortunately, sites state seems to be loaded by most routes, so there likely won't be any meaningful size savings, at least from modularising alone. It's an important step in making things less monolithic, however, and perhaps as preparation for flattening the state tree somewhat by moving some branches out of `state/sites/foo` to `state/foo`.

#### Changes proposed in this Pull Request

* Modularise sites state
* Minor lint fixes to touched files

#### Testing instructions

Since sites state seems to be loaded by most routes, there's no easy way to check that modularization is working. To ensure correctness, try to use site functionality as much as possible and see if anything breaks. The unit tests and E2E tests should add a good measure of confidence.
